lib/parsers/webp_parser.rb

```diff
@@ -119,5 +126,35 @@ def create_image(width, height, has_multiple_frames: false, has_transparency: fa
     )
   end
 
+  def augment_image(image)
+    # We're going to scan the file looking for the EXIF, XMP and/or ANMF chunks.
+    exif = nil
+    xmp = nil
+    num_frames = 0
+    loop do
+      chunk_code, chunk_size = safe_read(@buf, 8).unpack('A4V')
+      safe_skip(@buf, 1) if chunk_size.odd? # Padding byte of 0 added if chunk size is odd.
+      case chunk_code
+      when 'EXIF'
+        exif ||= exif_from_tiff_io(StringIO.new(safe_read(@buf, chunk_size)))
+      when 'XMP'
+        xmp ||= parse_xmp_metadata(chunk_size)
+      when 'ANMF'
+        num_frames += 1 if has_multiple_frames
+        safe_skip(@buf, chunk_size)
+      else
+        safe_skip(@buf, chunk_size)
+      end
+    end rescue FormatParser::IOUtils::InvalidRead
```

Comment:
could you move the `rescue FormatParser::IOUtils::InvalidRead` on a new line? we blind developers like good indentation or we tend to miss stuff 🤓

Reply:
that's correct! It wants to indent it there because it's where it should be :)
when you do:
```
def my_method
  do_something()
rescue ..
  do_something_else()
end
```

it's like you are doing:

```
def my_method
  begin
    do_something()
  rescue
    do_something_else()
  end
end
```

(we don't want the 2nd one because we want to use implicit `begin`s).